src/managers/GuildMemberManager.js

```diff
@@ -166,7 +167,7 @@ class GuildMemberManager extends BaseManager {
     const _data = { ...data };
     if (_data.channel) {
       _data.channel = this.guild.channels.resolve(_data.channel);
-      if (!_data.channel || _data.channel.type !== 'voice') {
+      if (!_data.channel || !(_data.channel instanceof BaseGuildVoiceChannel)) {
```

Comment:
`null instanceof Class` returns `false`, so the falsy check is unneeded.

```suggestion
      if (!(_data.channel instanceof BaseGuildVoiceChannel)) {
```